internal/gengapic/doc_file.go

```diff
@@ -203,6 +207,17 @@ func (g *generator) genDocFile(year int, scopes []string, serv *descriptor.Servi
 		p("}")
 		p("")
 	}
+
+	if hasREST {
+		p("// maybeUnknownEnum wraps the given proto-JSON parsing error if it is the result")
+		p("// of receiving an unknown enum value.")
+		p("func maybeUnknownEnum(err error) error {")
+		p(`  if strings.Contains(err.Error(), "invalid value for enum type") {`)
+		p(`    err = xerrors.Errorf("received an unknown enum value, a later version of the library may support it: %%w", err)`)
```

Comment:
nit: s/`,`/`;`/

Reply:
I'm not sure about reporting the line number if the JSON response is not available to the caller. Maybe we can omit it somehow? I did not know about xerrors, but it says "The returned error includes the file and line number of the caller when formatted *with additional detail enabled*" (emphasis added). It's not clear if that enablement is the `%w` specifically. I'm not quite grokking it yet.

WDYT? Regardless, probably not a blocker, though it would be nice to remove the line number.

Reply:
> nit: s/,/;/

Ack.

The message that comes after the `:` is the raw error message from the parser error and `%w` is the formatter that indicates the given error should be wrapped so that the original error can be `Unwrap`'d and retained. The `%w` in this case isn't adding any extra information, what is printed is just the output of `Error()` from the parser error. So I'd prefer not to mangle the original error by stripping parts of the message.

Reply:
Ah, I see. In that case, maybe we should wrap the error but format it differently? "If the format specifier includes a %w verb with an error operand in a position other than at the end, the returned error will still implement an Unwrap method returning the operand, but the error's Format method will not return the wrapped error." (https://pkg.go.dev/golang.org/x/xerrors#Errorf). IIUC, this would give us the error message  currently in the PR description up to the first `:`. WDYT?

(But no need to rat-hole on this)

Reply:
> IIUC, this would give us the error message currently in the PR description up to the first :. WDYT

Your understanding is correct, but then we wouldn't get the unknown enum value, which is part of the original error message. Is that an appropriate trade off? I'd rather not parse it out because we should just report the original wrapped error message (as we are now) instead.

Reply:
Oh, shoot, you're right. Displaying the unknown enum is probably more important than omitting the line number. Parsing the error is somewhat brittle, in case the text changes...and not worth doing now given the likelihood we'll adopt another solution.

OK, nm.